torchvision/models/densenet.py

```diff
@@ -35,41 +43,36 @@ def __init__(self, num_input_features, growth_rate, bn_size, drop_rate, memory_e
         self.drop_rate = float(drop_rate)
         self.memory_efficient = memory_efficient
 
-    def bn_function(self, inputs):
-        # type: (List[Tensor]) -> Tensor
+    def bn_function(self, inputs: List[Tensor]) -> Tensor:
         concated_features = torch.cat(inputs, 1)
         bottleneck_output = self.conv1(self.relu1(self.norm1(concated_features)))  # noqa: T484
         return bottleneck_output
 
     # todo: rewrite when torchscript supports any
-    def any_requires_grad(self, input):
-        # type: (List[Tensor]) -> bool
+    def any_requires_grad(self, input: List[Tensor]) -> Tensor:
         for tensor in input:
             if tensor.requires_grad:
                 return True
         return False
 
     @torch.jit.unused  # noqa: T484
-    def call_checkpoint_bottleneck(self, input):
-        # type: (List[Tensor]) -> Tensor
+    def call_checkpoint_bottleneck(self, input: List[Tensor]) -> Tensor:
         def closure(*inputs):
             return self.bn_function(inputs)
 
         return cp.checkpoint(closure, *input)
 
     @torch.jit._overload_method  # noqa: F811
-    def forward(self, input):
-        # type: (List[Tensor]) -> (Tensor)
+    def forward(self, input: List[Tensor]) -> Tensor:
         pass
 
     @torch.jit._overload_method  # noqa: F811
-    def forward(self, input):
-        # type: (Tensor) -> (Tensor)
+    def forward(self, input: Tensor) -> Tensor:
         pass
 
     # torchscript does not yet support *args, so we overload method
     # allowing it to take either a List[Tensor] or single Tensor
-    def forward(self, input):  # noqa: F811
+    def forward(self, input: Union[Tensor, List[Tensor]]) -> Tensor:  # noqa: F811
```

Comment:
`torch.script` is not able to handle `Union`. We need to annotate it as `torch.Tensor` for now and add an ignore for `mypy`.